From 00225fec6845779c0f3973eddc5357ca57bae22a Mon Sep 17 00:00:00 2001
From: Santosh Shukla <sshukla@marvell.com>
Date: Mon, 3 Sep 2018 20:54:03 +0530
Subject: [PATCH 0185/1051] soc: octeontx2: Fix mask for rvu_npc_install_xx

Now that kw_mask is used as full mask in below code snippet:

````
func: npc_get_keyword();
*cam1 &= kw_mask;
*cam0 = ~*cam1 & kw_mask;
````
So use mask for kw_mask[] in rvu_npc_install_{ucat/bcast/promisc} func.

Fixes: 7b4581a4c ("soc: octeontx2: Install ucast and bcast pkt forwarding rules")

Signed-off-by: Vivek Sharma <viveksharma@marvell.com>
Signed-off-by: Santosh Shukla <sshukla@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu_npc.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu_npc.c b/drivers/soc/marvell/octeontx2/rvu_npc.c
index dee6d2f09753..80cc953bee3f 100644
--- a/drivers/soc/marvell/octeontx2/rvu_npc.c
+++ b/drivers/soc/marvell/octeontx2/rvu_npc.c
@@ -328,11 +328,11 @@ void rvu_npc_install_ucast_entry(struct rvu *rvu, u16 pcifunc,
 
 	/* Match ingress channel and DMAC */
 	entry.kw[0] = chan;
-	entry.kw_mask[0] = ~chan & 0xFFFULL;
+	entry.kw_mask[0] = 0xFFFULL;
 
 	kwi = NPC_PARSE_RESULT_DMAC_OFFSET / sizeof(u64);
 	entry.kw[kwi] = mac;
-	entry.kw_mask[kwi] = ~mac & (BIT_ULL(48) - 1);
+	entry.kw_mask[kwi] = BIT_ULL(48) - 1;
 
 	/* Don't change the action if entry is already enabled
 	 * Otherwise RSS action may get overwritten.
@@ -371,12 +371,12 @@ void rvu_npc_install_promisc_entry(struct rvu *rvu, u16 pcifunc,
 					 nixlf, NIXLF_PROMISC_ENTRY);
 
 	entry.kw[0] = chan;
-	entry.kw_mask[0] = ~chan & 0xFFFULL;
+	entry.kw_mask[0] = 0xFFFULL;
 
 	if (allmulti) {
 		kwi = NPC_PARSE_RESULT_DMAC_OFFSET / sizeof(u64);
 		entry.kw[kwi] = BIT_ULL(40); /* LSB bit of 1st byte in DMAC */
-		entry.kw_mask[kwi] = 0x00;
+		entry.kw_mask[kwi] = BIT_ULL(40);
 	}
 
 	*(u64 *)&action = 0x00;
@@ -459,7 +459,7 @@ void rvu_npc_install_bcast_match_entry(struct rvu *rvu, u16 pcifunc,
 	 *
 	 **/
 	entry.kw[0] = BIT_ULL(13) | chan;
-	entry.kw_mask[0] = ~entry.kw[0] & (BIT_ULL(13) | 0xFFFULL);
+	entry.kw_mask[0] = (BIT_ULL(13) | 0xFFFULL);
 
 	*(u64 *)&action = 0x00;
 #ifdef MCAST_MCE
-- 
2.17.1

